packages/firestore/src/lite/settings.ts

```diff
@@ -28,12 +28,61 @@ import { validateIsNotUsedTogether } from '../util/input_validation';
 export const DEFAULT_HOST = 'firestore.googleapis.com';
 export const DEFAULT_SSL = true;
 
+/**
+ * Specifies custom configurations for your Cloud Firestore instance.
+ * You must set these before invoking any other methods.
+ */
 export interface Settings {
+  /** The hostname to connect to. */
   host?: string;
+
+  /** Whether to use SSL when connecting. */
   ssl?: boolean;
+
+  /**
+   * Whether to skip nested properties that are set to `undefined` during
+   * object serialization. If set to `true`, these properties are skipped
+   * and not written to Firestore. If set to `false` or omitted, the SDK
+   * throws an exception when it encounters properties of type `undefined`.
+   */
   ignoreUndefinedProperties?: boolean;
+
+  /**
+   * An approximate cache size threshold for the on-disk data. If the cache
+   * grows beyond this size, Firestore will start removing data that hasn't been
+   * recently used. The size is not a guarantee that the cache will stay below
+   * that size, only that if the cache exceeds the given size, cleanup will be
+   * attempted.
+   *
+   * The default value is 40 MB. The threshold must be set to at least 1 MB, and
+   * can be set to CACHE_SIZE_UNLIMITED to disable garbage collection.
+   */
   cacheSizeBytes?: number;
+
+  /**
+   * Forces the SDK’s underlying network transport (WebChannel) to use
+   * long-polling. Each response from the backend will be closed immediately
+   * after the backend sends data (by default responses are kept open in
+   * case the backend has more data to send). This avoids incompatibility
+   * issues with certain proxies, antivirus software, etc. that incorrectly
+   * buffer traffic indefinitely. Use of this option will cause some
```

Comment:
Global nit: don't use "will" unless you really mean future tense.  In places like this, present tense is fine.  "This option causes some performance..."